homeassistant/components/portainer/binary_sensor.py

```diff
@@ -0,0 +1,170 @@
+from .coordinator import PortainerCoordinator
+from .entity import (
+    PortainerContainerEntity,
+    PortainerCoordinatorData,
+    PortainerEndpointEntity,
+)
+
+
+@dataclass(frozen=True, kw_only=True)
+class PortainerBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Class to hold Portainer binary sensor description."""
+
+    state_fn: Callable[[Any], bool | None]
+    attributes_fn: Callable[[Any], dict[Any, StateType]] | None = None
+
+
+CONTAINER_SENSORS: tuple[PortainerBinarySensorEntityDescription, ...] = (
+    PortainerBinarySensorEntityDescription(
+        key="status",
+        translation_key="status",
+        state_fn=lambda data: bool(data.state == "running"),
+        device_class=BinarySensorDeviceClass.RUNNING,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+ENDPOINT_SENSORS: tuple[PortainerBinarySensorEntityDescription, ...] = (
+    PortainerBinarySensorEntityDescription(
+        key="status",
+        translation_key="status",
+        state_fn=lambda data: bool(
+            data.endpoint.status == 1
+        ),  # 1 = Running | 2 = Stopped
+        device_class=BinarySensorDeviceClass.RUNNING,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: PortainerConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Portainer binary sensors."""
+    portainer = entry.runtime_data.coordinator
+    entities: list[BinarySensorEntity] = []
+
+    for endpoint in portainer.endpoints.values():
+        entities.extend(
+            [
+                PortainerEndpointSensor(
+                    coordinator=portainer,
+                    entry=entry,
+                    entity_description=entity_description,
+                    device_info=endpoint,
+                )
+                for entity_description in ENDPOINT_SENSORS
+            ]
+        )
+
+        for container in endpoint.containers.values():
+            entities.extend(
+                [
+                    PortainerContainerSensor(
+                        coordinator=portainer,
+                        entry=entry,
+                        entity_description=entity_description,
+                        device_info=container,
+                        via_device=endpoint,
+                    )
+                    for entity_description in CONTAINER_SENSORS
+                ]
+            )
+
+    async_add_entities(entities, True)
+
+
+class PortainerEndpointSensor(PortainerEndpointEntity, BinarySensorEntity):
+    """Representation of a Portainer endpoint binary sensor entity."""
+
+    entity_description: PortainerBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PortainerCoordinator,
+        entry: PortainerConfigEntry,
+        entity_description: PortainerBinarySensorEntityDescription,
+        device_info: PortainerCoordinatorData,
+    ) -> None:
```

Comment:
not necessary to pass the config_entry, it is accessible via `self.coordinator.config_entry` and it should be sufficient to pass the device_id, the `device_info` object will be accessed via the coordinator 
```suggestion
    def __init__(
        self,
        coordinator: PortainerCoordinator,
        entity_description: PortainerBinarySensorEntityDescription,
        device_id: int,
    ) -> None:
```